src/Periphery/TokenWrapper.sol

```diff
@@ -30,18 +33,37 @@ contract TokenWrapper is WithdrawablePeriphery {
     // solhint-disable-next-line no-empty-blocks
     constructor(
         address _wrappedToken,
+        address _converter,
         address _owner
     ) WithdrawablePeriphery(_owner) {
-        wrappedToken = _wrappedToken;
-        IERC20(wrappedToken).approve(address(this), MAX_INT);
+        WRAPPED_TOKEN = _wrappedToken;
+        CONVERTER = _converter;
+        USE_CONVERTER = _converter != address(0);
+        WRAPPER_ADDRESS = USE_CONVERTER ? _converter : _wrappedToken;
```

Comment:
nitpick: feel this variable is unnecessary. We could just use CONVERTER (and in case we do not have a converter we also assign the _wrappedToken address to it). 

the variable CONVERTER is only used once in this whole contract (line 80) and that occurrence could arguably be replaced with WRAPPER_ADDRESS.